pkg/operator/ceph/pool/controller.go

```diff
@@ -335,7 +335,9 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 
 		// update ObservedGeneration in status at the end of reconcile
 		// Set Ready status, we are done reconciling
-		r.updateStatus(request.NamespacedName, cephv1.ConditionReady, observedGeneration)
+		if !r.updateStatus(request.NamespacedName, cephv1.ConditionReady, observedGeneration) {
```

Comment:
The reason for the current approach is that we prefer to continue the full reconcile even if the status fails to update. This way, when the status update fails, the pool will still complete the full reconcile. To get the benefits of your fix and the full reconcile, what if we store the error in a new `statusErr` variable, then continue the reconcile? Then at the end of the reconcile method, we requeue the reconcile due to the status update error. 
```suggestion
		statusErr = r.updateStatus(request.NamespacedName, cephv1.ConditionReady, observedGeneration) {
```